pkg/kubelet/reason_cache.go

```diff
@@ -40,17 +40,18 @@ type ReasonCache struct {
 	cache *lru.Cache
 }
 
-// reasonInfo is the cached item in ReasonCache
-type reasonInfo struct {
-	reason  error
-	message string
+// Reason is the cached item in ReasonCache
+type Reason struct {
```

Comment:
golint report a warning that the `error` has to be the latest one of return value at https://github.com/kubernetes/kubernetes/pull/47316/files#diff-ade44c0052b07470763ef431d55210ceL95 .
Compare to moving `error` to the latest one (the latest return value indicates whether found reason in cache), so prefer to use an object as the result value.